src/Windows.Core/DEVMODE.cs

```diff
@@ -0,0 +1,1326 @@
+﻿// Copyright © .NET Foundation and Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Runtime.InteropServices;
+
+    /// <summary>
+    /// Contains information about the initialization and environment of a printer or a display device.
+    /// </summary>
+    [StructLayout(LayoutKind.Explicit)]
+    public unsafe struct DEVMODE
+    {
+        public const int CCHDEVICENAME = 32;
+        public const int CCHFORMNAME = 32;
+
+        /// <summary>
+        /// A zero-terminated character array that specifies the "friendly" name of the printer or display; for example, "PCL/HP LaserJet" in the case of PCL/HP LaserJet.
+        /// This string is unique among device drivers. Note that this name may be truncated to fit in the <see cref="dmDeviceName"/> array.
+        /// </summary>
+        [FieldOffset(0)]
+        public fixed char dmDeviceName[CCHDEVICENAME];
+
+        /// <summary>
+        /// The version number of the initialization data specification on which the structure is based.
+        /// To ensure the correct version is used for any operating system, use <c>DM_SPECVERSION</c>.
+        /// </summary>
+        [FieldOffset(64)]
+        public ushort dmSpecVersion;
+
+        /// <summary>
+        /// The driver version number assigned by the driver developer.
+        /// </summary>
+        [FieldOffset(66)]
+        public ushort dmDriverVersion;
+
+        /// <summary>
+        /// Specifies the size, in bytes, of the <see cref="DEVMODE"/> structure, not including any private driver-specific data that might follow the structure's public members
+        /// Set this member to size of <see cref="DEVMODE"/> to indicate the version of the <see cref="DEVMODE"/> structure being used.
+        /// </summary>
+        [FieldOffset(68)]
+        public ushort dmSize;
+
+        /// <summary>
+        /// Contains the number of bytes of private driver-data that follow this structure.
+        /// If a device driver does not use device-specific information, set this member to zero.
+        /// </summary>
+        [FieldOffset(70)]
+        public ushort dmDriverExtra;
+
+        /// <summary>
+        /// Specifies whether certain members of the <see cref="DEVMODE"/> structure have been initialized.
+        /// If a member is initialized, its corresponding bit is set, otherwise the bit is clear. A driver supports only those <see cref="DEVMODE"/> members that are appropriate for the printer or display technology.
+        /// </summary>
+        [FieldOffset(72)]
+        public DEVMODE_FieldUseFlags dmFields;
```

Comment:
Since `DEVMODE[W]` is a `struct`, could structuring this (and other types where this suggestion might apply as well) as a nested-type `DEVMODE.FieldUseFlags` make this a bit cleaner to read/use ? 

I'm usually bit wary of nested public types. They are impossible to type-forward to another assembly. That said, this repo is designed around nested public types, so I figured I'd offer-up this suggestion 😉

Reply:
I think might be too risky, no?
`DEVMODE` is a bit central. It's also used around printers and other devices.

Reply:
My question was based on an inference off of the type's name - `DEVMODE_<TypeName>`- which to me suggests that the type is *atleast somewhat* exclusive to `DEVMODE`. If it's indeed exclusive, then nesting would not lead to discovery problems (and it won't affect the layout of the `DEVMODE` struct AFAIK - you can test it). 

OTOH, if these types don't have a strong affinity to `DEVMODE`, then perhaps they don't have to be named this way. They could just be named `<Typename>` and added to the `Windows.Core` `PInvoke` namespace directly.

Reply:
It's going to be hard to track down all those types and where they're used in.
There are these two things to consider:
- Skimming through the _wingdi.h_ header (where `DEVMODE` is defined), it doesn't seem there are any other usages for these enums. I don't have the time resources to make proper research to conclude otherwise.
- They are defined globally in the header, so we might consider exposing them as such too.

Reply:
Header definitions are all going to be global. Windows headers conventionally everything at a flat global scope. 

If these must be global-ish, then perhaps consider using C# conventions for type names - esp. since these names aren't directly lifted from a Windows header. For e.g., s/`DEVMODE_AbcFlags`/`DevModeAbcFlags`, etc.

Reply:
I'm not sure what's the final verdict here.
They're currently all under the `PInvoke` namespace, but they're all called `DEVMODE_AbcFlags`. Do you want me to change them all?

Reply:
My comments are suggestions, not requirements 😄 

Where these types go - `PInvoke` namespace vs. nested type inside `DEVMODE` struct is I think a judgment call. I'd defer to your sensibility here since you have researched this plenty. If you think they are better suited under the `PInvoke` namespace then that's where they should go IMHO 😄 

Re: How to name these types: `DEVMODE_EnumName` vs. `DevModeTypeName` - is a matter of style and consistency. There are arguments that support either style. `DEVMODE_EnumName` is an obvious choice since `DEVMODE` is how the Win32 type is styled and using that as a prefix aids in discovery; OTOH `DevModeTypeName` is how C# types are typically styled, and types like [DisplayDeviceFlags](https://github.com/dotnet/pinvoke/blob/master/src/Windows.Core/DISPLAY_DEVICE%2BDisplayDeviceFlags.cs) use this convention (instead of say, naming the enum type [`DISPLAY_DEVICE_StateFlags`](https://github.com/dotnet/pinvoke/blob/86376b1de31a81507b6f6a00aa8d8868fc94665b/src/Windows.Core/DISPLAY_DEVICE.cs#L29))

Reply:
Awesome then. The way they're all currently named is `PInvoke.DEVMODE_SomeOptionFlags`. And BTW I wrote the `DISPLAY_DEVICE_StateFlags` 😍, but I did it based on other occurrences I found in PInvoke.
So I guess we're probably good to go, unless I missed something.

Reply:
> I'm usually bit wary of nested public types. They are impossible to type-forward to another assembly. That said, this repo is designed around nested public types, so I figured I'd offer-up this suggestion 😉

See https://github.com/dotnet/pinvoke/issues/288#issuecomment-652750274. Although we've traditionally put all structs as nested under the library class, I think we should reverse that design and put *most* structs directly under `PInvoke`, including most existing ones in a breaking API change when we're ready to make such a change. For now, new structs can follow the new policy, IMO. 
However, where a struct or enum is so coupled to another struct such that you'd name one after the other, I think nesting makes sense. It just changes `DEVMODE_SomeOptionFlags` to `DEVMODE.SomeOptionFlags` and in the meantime removes `*SomeOptionFlags` from the completion lists of other scenarios, which seems like goodness, IMO. 

That leaves the question of whether it should simply be `SomeOptionsFlags` vs. `DEVMODE*SomeOptionFlags`. For that, as has also been expressed here, if we knew that `SomeOptionFlags` was actually used by other APIs, I'd favor making them more prominently named/placed. But if we're reasonably confident that it only applies to one struct, I'd keep it named as if it only applied there.
If the worst happened (i.e. we discovered another API that uses it too), we can always simply define another enum with the same values but with a more prominent name.

Reply:
Looking over this in general, this leads me to believe the only change to make here is to move these `DEVMODE_` types into `DEVMODE` as nested types and trim the prefix from them, since their existing names already commit us to consider these tightly coupled.